lib/tapioca/dsl/compilers/active_record_relations.rb

```diff
@@ -234,6 +228,38 @@ def gather_constants
 
         private
 
+        sig { returns(RBI::Scope) }
+        def model
+          @model ||= T.let(
```

Comment:
For some time now, we create an instance of the compiler per-constant, so that the compiler can keep internal state like this. That's why you can rely on the `constant` attr reader to work properly inside the compiler.

Since each instance of the compiler only processes a single constant, it is totally fine for it to cache the related `model` that corresponds to that constant as well.

Reply:
https://github.com/Shopify/tapioca/blob/main/lib/tapioca/dsl/compiler.rb#L60-L65

Reply:
Oh awesome! Thanks for the context.